### PR TITLE
Fact 677 - Add in new 'Local Authorities' editing tab

### PR DESCRIPTION
**Pull request by Majid-justice**
https://tools.hmcts.net/jira/browse/FACT-677

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/controller/lists/LocalAuthoritiesListController.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+      const localAuthorityItems = localAuthorities.map((la: LocalAuthority) => (
+        {
+          id: la.name.replace(/ /g,'_'),
+          value: JSON.stringify(la),
+          text: la.name,
+          checked : selectedLocalAuthority ? (la.id === selectedLocalAuthority.id) : false
+        }));
+
+      return localAuthorityItems.sort((a, b) => (a.text < b.text ? -1 : 1));
```
Comment (KianKwa) [inline src/main/app/controller/lists/LocalAuthoritiesListController.ts:87]: Can we not use sort((a,b) => a.text - b.text)? This returns negative if a smaller than b and will return positive if a bigger than b. Does it not return the same result?

File: src/test/functional/features/list-local-authorities.feature [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Examples:
+      | local authority id     | local authority        |
+      | Barnet_Borough_Council | Barnet Borough Council |
+
+  Scenario Outline: As a user when I try to enter duplicate local authority it should not allow me to update and acknowledge me with proper error message
+
+    When I select local authority "<local authority id>"
+    When I edit the local authority "<duplicate local authority>"
+    When I click on save local authority list
+    Then An error is displayed for edit local authorities with title "There is a problem" and summery "Local Authority already exists."
```
Comment (KianKwa) [inline src/test/functional/features/list-local-authorities.feature:32]: typo here should be 'summary' not 'summery'

File: src/test/functional/features/list-local-authorities.feature [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    Examples:
+      | local authority id     | duplicate local authority |
+      | Barnet_Borough_Council | Luton Borough Council     |
+
+  Scenario Outline: As a user when I try to enter invalid local authority it should not allow me to update and acknowledge me with proper error message
+
+    When I select local authority "<local authority id>"
+    When I edit the local authority "<invalid local authority>"
+    When I click on save local authority list
+    Then An error is displayed for edit local authorities with title "There is a problem" and summery "Invalid Local Authority entered."
```
Comment (KianKwa) [inline src/test/functional/features/list-local-authorities.feature:43]: typo here should be 'summary' not 'summery'

File: src/test/functional/features/list-local-authorities.feature [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    Examples:
+      | local authority id     | invalid local authority  |
+      | Barnet_Borough_Council | Lutonnnc Borough Council |
+
+  Scenario Outline: As a user when I try to enter blanc local authority field it should not allow me to update and acknowledge me with proper error message
+
+    When I select local authority "<local authority id>"
+    When I edit the local authority ""
+    When I click on save local authority list
+    Then An error is displayed for edit local authorities with title "There is a problem" and summery "The local authority name is required."
```
Comment (KianKwa) [inline src/test/functional/features/list-local-authorities.feature:54]: typo here should be 'summary' not 'summery'

**Review by KianKwa — APPROVED** [review]

File: src/main/app/controller/lists/LocalAuthoritiesListController.ts (hunk 1 above)
Comment (Majid-justice) [reply src/main/app/controller/lists/LocalAuthoritiesListController.ts:87]: tried using your suggestion return syntax "The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type"

File: src/test/functional/features/list-local-authorities.feature (hunk 2 above)
Comment (Majid-justice) [reply src/test/functional/features/list-local-authorities.feature]: amended

File: src/test/functional/features/list-local-authorities.feature (hunk 3 above)
Comment (Majid-justice) [reply src/test/functional/features/list-local-authorities.feature]: amended

File: src/test/functional/features/list-local-authorities.feature (hunk 4 above)
Comment (Majid-justice) [reply src/test/functional/features/list-local-authorities.feature]: amended

**Pull request merged**